keylime/common/algorithms.py

```diff
@@ -12,47 +12,34 @@ def is_accepted(algorithm: str, accepted: List[Any]) -> bool:
     return algorithm in accepted
 
 
-def _hashit(algorithm: str, data: bytes) -> bytes:
-    if algorithm == "sha1":
-        return hashlib.sha1(data).digest()
-    if algorithm == "sha256":
-        return hashlib.sha256(data).digest()
-    if algorithm == "sha384":
-        return hashlib.sha384(data).digest()
-    if algorithm == "sha512":
-        return hashlib.sha512(data).digest()
-    if algorithm == "sm3_256":
-        # SM3 is not guaranteed to be there
-        return hashlib.new("sm3", data).digest()
-
-    raise ValueError(f"Unsupported hash algorithm {algorithm}")
-
-
 class Hash(str, enum.Enum):
+    # Names compatible with tpm2-tools (man/common/alg.md)
     SHA1 = "sha1"
     SHA256 = "sha256"
     SHA384 = "sha384"
     SHA512 = "sha512"
     SM3_256 = "sm3_256"
 
-    def __init__(self, *args):  # pylint: disable=unused-argument
-        super().__init__()
-        # Test hash to raise ValueError for unsupported hashes
-        _hashit(self.value, b"")
+    @classmethod
+    def from_algorithm(cls, algorithm: str) -> "Hash":
+        return cls(algorithm)
 
     @staticmethod
     def is_recognized(algorithm: str) -> bool:
         try:
-            Hash(algorithm)
-            return True
+            Hash(algorithm).get_size()
         except ValueError:
             return False
+        return True
+
+    def __hashfn(self, data: bytes) -> Any:
+        return hashlib.new(self.value, data)
```

Comment:
This will fail with `sm3_256` because in openssl it is only called `sm3`.

Reply:
Right, I extracted the function just to address this issue : )

Done!